api/src/main/java/net/kyori/adventure/text/Component.java

```diff
@@ -130,10 +130,12 @@
    * @param components the components
    * @return a text component
    * @since 4.0.0
+   * @deprecated for removal since 4.8.0, use {@link #join(JoinConfiguration, ComponentLike...)} with a {@link JoinConfiguration} instead.
    */
+  @Deprecated
```

Comment:
I don't think we should deprecate this method - it is perfectly fine as a convenient shorthand.

Reply:
Text replacement methods got the same treatment - deprecated in favour of configuration. It is easier to maintain `JoinConfiguration` and not worrying about updating things here too.

Reply:
This brings up a good point -- perhaps we should un-deprecate `replaceText` methods for the most simple case of Pattern -> Component, as a convenience option.

Reply:
Another solution is to just add convenience methods to `TextReplacementConfig` similar to the `separator` methods in `JoinConfiguration`, for example.

Reply:
hm, I think if we were doing this from the start that would be good, but in the interest of minimizing API churn, I think maintaining the older methods would be the better way to go for both this and replacement.

Reply:
Thinking about this again, and with #418 I think replacing all these methods with one that takes a `JoinConfiguration` is the least awkward way forward.